src/main/java/org/openrewrite/java/migrate/jakarta/UpdateBeanManagerMethod.java

```diff
@@ -0,0 +1,97 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate.jakarta;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.jetbrains.annotations.NotNull;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.JavaParser;
+
+public class UpdateBeanManagerMethod extends Recipe {
+    @Option(displayName = "Method Pattern",
+            description = "A `BeanManager.fireEvent()` or `BeanManager.createInjectionTarget()` matching required",
+            example = "jakarta.enterprise.inject.spi.BeanManager fireEvent()")
+    @NonNull String methodPattern;
+
+    @Override
+    public @NotNull String getDisplayName() {
+        return "Update `fireEvent()` and `createInjectionTarget()` calls";
+    }
+
+    @Override
+    public @NotNull String getDescription() {
+        return " Updates `BeanManager.fireEvent()` or `BeanManager.createInjectionTarget()`";
+    }
+
+    MethodMatcher methodInputPattern = null;
+
+    @JsonCreator
+    public UpdateBeanManagerMethod(@NonNull @JsonProperty("methodPattern") String methodPattern) {
+        this.methodPattern = methodPattern;
+        methodInputPattern = new MethodMatcher(methodPattern, false);
+    }
+
+    @Override
+    public @NotNull TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MethodInvocationVisitor(methodPattern);
+    }
+
+    private static class MethodInvocationVisitor extends JavaVisitor<ExecutionContext> {
+        private final MethodMatcher methodPattern;
+
+        private MethodInvocationVisitor(String methodPattern) {
+            this.methodPattern = new MethodMatcher(methodPattern, false);
+        }
+
+        @Nullable
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ec) {
+            J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, ec);
+            if (methodPattern.matches(method)) {
+                String newMethodName = "";
+                if (method.getSimpleName().equals("fireEvent")) {
+                    newMethodName = "getEvent()." + "fire";
+
+                    JavaType.Method type = method.getMethodType();
+                    if (type != null) {
+                        type = type.withName(newMethodName);
+                    }
+                    return method.withName(method.getName().withSimpleName(newMethodName)).withMethodType(type);
```

Comment:
This won't work well with our type system I think, to have a `methodName` containing `getEvent().fire`. Instead we should likely use a JavaTemplate here to create the chain of method invocations.

Reply:
This was also not addressed.

Reply:
@timtebeek I tried to update the code with the template..no matter what I tried it keeps failing.Hence I used this way ```
//                    maybeRemoveImport("jakarta.enterprise.inject.spi.BeanManager");
//                    return JavaTemplate.builder("#{any(jakarta.enterprise.inject.spi.BeanManager)}.getEvent().fire(#{any()})")
//                            .build()
//                            .apply(updateCursor(mi), mi.getCoordinates().replace(), mi.getSelect(), mi.getArguments().get(0));
```
Not sure why the template approach didnt work for getEvent()